app/src/main/java/chat/rocket/android/helper/MessageParser.kt

```diff
@@ -0,0 +1,177 @@
+    }
+
+    private fun applySpans(text: CharSequence) {
+        val matcher = userPattern.matcher(text)
+        val result = text as Spannable
+        while (matcher.find()) {
+            val user = matcher.group(1)
+            val start = matcher.start()
+            //TODO: should check if username actually exists prior to applying.
+            result.setSpan(UsernameClickableSpan(), start, start + user.length, 0)
+        }
+    }
+
+    /**
+     * Convert to a lenient markdown consistent with Rocket.Chat web markdown instead of the official specs.
+     */
+    private fun toLenientMarkdown(text: String): String {
+        return text.trim().replace("\\*(.+)\\*".toRegex()) { "**${it.groupValues[1].trim()}**" }
+                .replace("\\~(.+)\\~".toRegex()) { "~~${it.groupValues[1].trim()}~~" }
+                .replace("\\_(.+)\\_".toRegex()) { "_${it.groupValues[1].trim()}_" }
+    }
+
+    class QuoteMessageSenderVisitor(private val context: Context,
+                                    configuration: SpannableConfiguration,
+                                    private val builder: SpannableBuilder,
+                                    private val senderNameLength: Int) : SpannableMarkdownVisitor(configuration, builder) {
+
+        override fun visit(blockQuote: BlockQuote) {
+
+            // mark current length
+            val length = builder.length()
+
+            // pass to super to apply markdown
+            super.visit(blockQuote)
+
+            val res = context.resources
+            val timeOffsetStart = length + senderNameLength + 1
+            builder.setSpan(QuoteMarginSpan(context.getDrawable(R.drawable.quote), 10), length, builder.length())
+            builder.setSpan(StyleSpan(Typeface.BOLD), length, length + senderNameLength)
+            builder.setSpan(ForegroundColorSpan(Color.BLACK), length, builder.length())
+            // set time spans
+            builder.setSpan(AbsoluteSizeSpan(res.getDimensionPixelSize(R.dimen.message_time_text_size)),
+                    timeOffsetStart, builder.length())
+            builder.setSpan(ForegroundColorSpan(res.getColor(R.color.darkGray)),
+                    timeOffsetStart, builder.length())
+        }
+    }
+
+    class QuoteMessageBodyVisitor(private val context: Context,
+                                  configuration: SpannableConfiguration,
+                                  private val builder: SpannableBuilder) : SpannableMarkdownVisitor(configuration, builder) {
+
+        override fun visit(blockQuote: BlockQuote) {
+
+            // mark current length
+            val length = builder.length()
+
+            // pass to super to apply markdown
+            super.visit(blockQuote)
+
+            builder.setSpan(QuoteMarginSpan(context.getDrawable(R.drawable.quote), 10), length, builder.length())
+        }
+    }
+
+    class QuoteMarginSpan(b: Drawable, pad: Int) : LeadingMarginSpan, LineHeightSpan {
+        private var mDrawable: Drawable? = b
```

Comment:
why nullable if the constructor's Drawable is nonnull.
```kotlin
class QuoteMarginSpan(private val drawable, private val pad)
```
And please, no Hungarian notation :pray:

Reply:
Nice catch. And sorry it was unintentional. That's and update version of another span. I'll fix it. :+1: